### DISCUSSION TITLE
Module complication fails

### DISCUSSION THREAD

**Discussion by FyiurAmron** [nxrighthere/UnrealCLR#209] [Help] [answered]
Tried both manual and automatic installs, and both 1.30 & edge versions, both fail with:

![image](https://user-images.githubusercontent.com/11016998/142735364-51959900-dc0b-47f3-80c3-b9a9074b2e15.png)

![image](https://user-images.githubusercontent.com/11016998/142735372-131e9b49-ec47-454f-97e2-300dca31f211.png)

I don't have the C++ toolchain set up in this editor instance (and I guess that it is required to rebuild the plugin), but was still wondering if it's ever possible to use the existing binary distribution in any way, without manual rebuild?

**Comment by nxrighthere** [comment]
External code in your project prevents the compilation of the plugin's module. Please, check the building log and resolve the issue in the project.

**Reply by nxrighthere** [reply]
Similar problem https://github.com/nxrighthere/UnrealCLR/discussions/94#discussioncomment-1247180

**Comment by FyiurAmron** [comment, marked as answer]
>External code in your project prevents the compilation of the plugin's module. Please, check the building log and resolve the issue in the project.
Hmm. Not a case, since it was a clean project (no 3rd party plugins, no nothing actually), but you got me thinking. I checked the logs and I got

>C:/Codes/UE_5.0EA/Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.exe Development Win64 -Project="C:/Codes/UE_5.0EA/_Projecs/TestBlueprintPuzzle/TestBlueprintPuzzle.uproject" -TargetType=Editor -Progress -NoEngineChanges -NoHotReloadFromIDE
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '3.1.0' (x64) was not found.

there. I realized that it's a problem with `Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.exe` , which is dependent on .NET Core 3.1 ATM, but emits no UI information about it nor installs it by default. I'll close this issue as it's not fixable on UnrealCLR side, hope people with the same problem may benefit from this knowledge. (tl;dr both C++ build tools *and* .NET Core 3.1 runtime is required beforehand, apart from the matching SDK for UnrealCLR)

@nxrighthere may I suggest adding this (as e.g.
* [.NET Core 3.1 runtime](https://dotnet.microsoft.com/download/dotnet/3.1/runtime?utm_source=getdotnetcore&utm_medium=referral) (required by UnrealBuildTool.exe)
) to the project readme in Building|Prerequisites section), since https://docs.unrealengine.com/en-US/Programming/Development/VisualStudioSetup/index.html#runtheunrealenginepre-requisiteinstaller doesn't cover this at all?

**Reply by nxrighthere** [reply]
This is something that Epic should add to their docs. Feel free to [contact](mailto:indies@unrealengine.com) them.
